aws-analytics-pinpoint/src/androidTest/java/com/amplifyframework/analytics/pinpoint/AnalyticsPinpointInstrumentedTest.java

```diff
@@ -138,6 +140,50 @@ public void testAutoFlush() throws AnalyticsException {
         assertEquals(0, analyticsClient.getAllEvents().size());
     }
 
+    /**
+     * Registers a global property and ensures that all recorded events have the global property.
+     * Unregisters a global property and ensures that it is respected in events recorded thereafter.
+     * @throws AnalyticsException Caused by incorrect usage of the Analytics API.
+     * @throws JSONException Caused by unexpected event structure.
+     */
+    @Test
+    public void testGlobalProperties() throws AnalyticsException, JSONException {
```

Comment:
if possible refactor this test into multiple smaller tests where each tests asserts on one major condition.

Reply:
There are just two tests here registering and unregistering global properties. Unregistering is better tested with some global properties registered, so thought of joining them together.